lib/HLSL/DxilTypeSystem.cpp

```diff
@@ -156,6 +156,39 @@ const Function *DxilFunctionAnnotation::GetFunction() const {
   return m_pFunction;
 }
 
+DxilFunctionFPFlag &DxilFunctionAnnotation::GetFlag() {
+  return m_fpFlag;
+}
+
+const DxilFunctionFPFlag &DxilFunctionAnnotation::GetFlag() const {
+  return m_fpFlag;
+}
+
+//------------------------------------------------------------------------------
+//
+// DxilFunctionFPFlag class methods.
+//
+
+void DxilFunctionFPFlag::SetFP32DenormMode(const DXIL::FPDenormMode mode) {
+  m_flag |= (int32_t)mode;
+}
+
+DXIL::FPDenormMode DxilFunctionFPFlag::GetFP32DenormMode() {
+  return (DXIL::FPDenormMode)(m_flag&0x7);
```

Comment:
Can we define a constant for the mask value (either in the class, or just above the Set... function) so the detail isn't embedded in the code here?  Then you can also apply the mask in SetFP32DenormMode: `m_flag |= kFPDenormMask & (int32_t)mode`, or DXASSERT that no other bits are set there.